dev_guide/application_lifecycle/promoting_applications.adoc

```diff
@@ -0,0 +1,781 @@
+operations that effectively move applications between environments, e.g., by
+performing an image tag operation from within a hook. However, the various hook
+points are best suited to managing an application's lifecycle within a given
+environment (e.g., using them to perform database schema migrations when a new
+version of the application is deployed), rather than to move application
+components between environments.
+====
+
+[[dev-guide-promoting-application-managing-api-objects]]
+=== Managing API Objects
+
+Resources, as defined in one environment, will be exported as JSON or YAML file
+content in preparation for importing it into a new environment. Therefore, the
+expression of API objects as JSON or YAML serves as the unit of work as you
+promote API objects through your application pipeline. The `oc` CLI is used to
+export and import this content.
+
+[TIP]
+====
+While not required for promotion flows with {product-title}, with the JSON or
+YAML stored in files, you can consider storing and retrieving the content from a
+SCM system. This allows you to leverage the versioning related capabilities of
+the SCM, including the creation of branches, and the assignment of and query on
+various labels or tags associated to versions.
+====
+
+[[dev-guide-promoting-application-exporting-api-object-state]]
+==== Exporting API Object State
+
+API object specifications should be captured with `oc export`. This operation
+removes environment specific data from the object definitions (e.g., current
+namespace or assigned IP addresses), allowing them to be recreated in different
+environments (unlike `oc get` operations, which output an unfiltered state of
+the object).
+
+Use of `oc label`, which allows for adding, modifying, or removing labels on API
+objects, can prove useful as you organize the set of object collected for
+promotion flows, because labels allow for selection and management of groups of
+pods in a single operation. This makes it easier to export the correct set of
+objects and, because the labels will carry forward when the objects are created
+in a new environment, they also make for easier management of the application
+components in each environment.
+
+[NOTE]
+====
+API objects often contain references such as a `DeploymentConfig` that
+references a `Secret`. When moving an API object from one environment to
+another, you must ensure that such references are also moved to the new
+environment.
+
+Similarly, API objects such as a `DeploymentConfig` often contain references to
+`ImageStreams` that reference an external registry. When moving an API object
+from one environment to another, you must ensure such references are resolvable
+within the new environment, meaning that the reference must be resolvable and
+the `ImageStream` must reference an accessible registry in the new environment.
+See the section on moving images as well as the section on promotion caveats for
```

Comment:
See the sections on  xref:dev-guide-promoting-applications-moving-images[Moving Images] and xref:dev-guide-promoting-applications-caveats[promotion caveats] for more detail.